From f54b58972c31d21738dc1c2fcc41e3d750bf30c5 Mon Sep 17 00:00:00 2001
From: Catalin Udma <catalin-dan.udma@nxp.com>
Date: Wed, 30 Jun 2021 17:25:11 +0300
Subject: [PATCH 23/29] s32:default environment: fix default environment when
 XEN_SUPPORT is enabled

Avoid defining bootcmd in two places when XEN is enabled
The issue is observed when scripts/get_default_envs.sh is executed:
two definitions of bootcmd are stored in object file's
.rodata.default_environment section, thus the script generated
environment contains both. When this environment is enabled on the
target, the latest one overwrites the corect command.

Issue: ALB-7261

Signed-off-by: Catalin Udma <catalin-dan.udma@nxp.com>
---
 include/configs/s32.h | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/include/configs/s32.h b/include/configs/s32.h
index 70d1f9a3e6..105555ff4c 100644
--- a/include/configs/s32.h
+++ b/include/configs/s32.h
@@ -399,8 +399,9 @@
 		"fdt rm /chosen linux,initrd-end; \0" \
 	"updatexenfdt=run chosen_node_setup; run dom0_node_setup; " \
 		"run bootargs_setup; run clear_default_chosen;\0" \
-	"bootcmd=" XEN_LOAD_FILES "run updatexenfdt; " \
-		"booti ${loadaddr} - ${fdt_addr}\0"
+
+#define XEN_BOOTCMD \
+	XEN_LOAD_FILES "run updatexenfdt; booti ${loadaddr} - ${fdt_addr}"
 #else
 #define XEN_EXTRA_ENV_SETTINGS  ""
 #endif
@@ -547,6 +548,9 @@
 #  define CONFIG_BOOTCOMMAND \
 	PFE_INIT_CMD "run flashboot"
 #elif defined(CONFIG_SD_BOOT)
+#ifdef CONFIG_XEN_SUPPORT
+#  define CONFIG_BOOTCOMMAND XEN_BOOTCMD
+#else
 #  define CONFIG_BOOTCOMMAND \
 	PFE_INIT_CMD "mmc dev ${mmcdev}; if mmc rescan; then " \
 		   "if run loadimage; then " \
@@ -555,6 +559,7 @@
 		   "fi; " \
 	   "else run netboot; fi"
 #endif
+#endif
 
 /* Miscellaneous configurable options */
 #define CONFIG_SYS_PROMPT_HUSH_PS2	"> "
-- 
2.17.1

